### DISCUSSION TITLE
Is this the normal behavior of the "total" property?

### DISCUSSION THREAD

**Discussion by vcapp** [appwrite/appwrite#4525] [Q&A]
Hi!

In the [documentation](https://appwrite.io/docs/models/documentList) it says that **total** is the  "Total number of documents documents that matched your query." (Yes 'documents' was written twice).

In most cases I would agree with the description. For example when I use the **equal** query to get a document with a specific **$id**, my total count is of course **1**. But what is for example with limit. Shouldn't a limit of 10 set the total count to 10 (For a collection with more than 10 documents)? Instead the total count is set to the number of all documents. And also when using cursorAfter or cursorBefore the total property is always equal to the number of all documents.

Thanks for an answer and keep up the good work! I love appwrite!

**Comment by stnguyen90** [comment]
Better wording would probably be:

> Total number of documents that matched your filter up to 5000

Does that answer your question?

cc @gewenyu99
